File: packages/zynth-filesystem/ios/ZynthFileSystem.swift
//
//  ZynthFileSystem.swift
//  ZynthFileSystem
//
//  Public interface for ZynthFileSystem module
//

import Foundation
import ZynthKit

@objc(ZynthFileSystem)
public class ZynthFileSystem: NSObject {
  private static var moduleInstance: ZynthFileSystemModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthFileSystem] Module already initialized")
      return
    }

    let module = ZynthFileSystemModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthFileSystem] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/zynth-filesystem/android/ZynthFileSystem/src/main/java/dev/zynth/filesystem/ZynthFileSystem.kt
package dev.zynth.filesystem

import android.app.Activity
import android.util.Log
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthFileSystem module.
 */
object ZynthFileSystem {
    private var moduleInstance: ZynthFileSystemModule? = null

    /**
     * Initialize the FileSystem module with an Activity and ZynthRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            Log.w("ZynthFileSystem", "Module already initialized")
            return
        }

        moduleInstance = ZynthFileSystemModule(activity.applicationContext)
        runtime.installModules(listOf(moduleInstance!!))
        Log.d("ZynthFileSystem", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}
